Gordo redesign and dedede adjustments

SirTanknSpank opened the pull request:
Big adjustments for the big penguin and his funny spike ball to make him less gordo centric, but keeping gordo as a incredibly low knockback tool that sets up dedede for combos or pressure. Oh and a few other things too.

Assets: [dedede-assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/11630786/dedede-assets.zip)



# Changelog

### Gordo Throw

- [!] New animations, based off of brawl's waddle dee toss
- [-] Hammer hitbox removed
- [-] Regrab removed
- [+] Startup: Frame 29 --> 18
- [+] FAF: Frame 58 --> 49
- [!] Can now ~~waddle~~dash by holding shield while doing side b on the ground
    -  This can slide off of ledges




https://github.com/HDR-Development/HewDraw-Remix/assets/115139386/0eea5fe2-3726-4d54-a997-978134a29624






https://github.com/HDR-Development/HewDraw-Remix/assets/115139386/20252a59-8585-46c2-9804-47611975ce32






https://github.com/HDR-Development/HewDraw-Remix/assets/115139386/4ba20163-9aa0-4969-8687-76e6a8e601d7








### Gordo (the projectile)
- [!] Damage is based off of how many times it bounces, rather than how long it has been active
- [-] Damage: 14/12.5/11/9.5 (changing every 30 frames) --> 10/9/8/7 (changing on every bounce)
- [+] KBG: 120
- [-] BKB: 0
- [+] FKB: 60
- [+] Vanishes upon being hit with an attack doing 5% or greater, rather then being sent back with a hitbox
    - This means that Gordo can no longer hit Dedede unless reflected or parried
- [-] Bounce Height Multiplier: 1.15x --> 1.05x
- [-] Hop life (the state gordo goes into after hitting someone): 60 -> 45 

### Gordo Shot (Inhaled Gordo)
- [!] Damage is based off of how many times it bounces, rather than how long it has been active
- [-] Damage: 16.8/15.0/13.2/11.4 (changing every 30 frames) --> 12.8/11.2/9.6/8.0 (changing on every bounce)
- [-] Angle: 40 --> 60
- [-] KBG: 66 --> 50
- [-] BKB: 70 --> 60

### Down Tilt
- [!] New animation
- [-] Startup: 6 --> 7
- [+] Angle: 40 --> 361
- [+] Initial Hit KBG: 80 --> 100
- [+] Late Hit KBG: 80 --> 100
- [-] Initial Hit BKB: 50 --> 30
- [-] Late Hit BKB: 50 --> 30
- [-] Late Hit damage: 8.0 --> 6.0


https://github.com/HDR-Development/HewDraw-Remix/assets/115139386/3316435f-a0a7-47ce-b5db-83a57f38918e



### Forward Air
- [-] KBG: 100 --> 94
- [~] Now a position based hitbox
  - This enables F-air to reverse hit
### Inhale
- [-] End Lag: 19 --> 25

### Forward Smash
- [+] Sweetspot Shieldstun Multiplier: 1.0x -> 1.5x

### Side Taunt
- [+] Fixed the bugged animation from vanilla ultimate

SirTanknSpank committed:
Initial Dedede Adjustments

SirTanknSpank committed:
Sound/Effect/Expression changes

- Adds proper effects for side b and down tilt
- Adds better sound for side b
- edited expression scripts for sideb and down tilt
- training mode reset for gordo

SirTanknSpank committed:
Changes from feedback

- Implemented gordo stock system
- 3 gordos, then goes on cd for 10 seconds
- Slight dtilt hitbox changes

SirTanknSpank committed:
Reverting Stock System/Fsmash change

-Reverted back to cooldown system, now 2.5 seconds (150 frames)
-Added shieldstun multiplier to f smash

hdr-balance-backroom reviewed (dismissed):
- Fair should be position-based
- Revert fair autocancel window
- Consider removing the lockout on Gordos as they don't seem to warrant it

SirTanknSpank committed:
Backroom changes

- Changing Fair to be position based
- Reverting AC window
- Removing Gordo Timer

SkewedAskew commented on fighters/common/src/function_hooks/transition.rs:
```diff
@@ -177,6 +177,14 @@ unsafe fn is_enable_transition_term_hook(boma: &mut BattleObjectModuleAccessor,
         }
 
+        //Disables Dedede Side Special on a timer
+        if boma.kind() == *FIGHTER_KIND_DEDEDE {
```
This can be removed as well

SkewedAskew commented on dynamic/src/consts.rs:
```diff
@@ -350,6 +350,12 @@ pub mod vars {
 
+    pub mod dedede {
+        pub mod instance {
+            pub const GORDO_TIMER: i32 = 0x0100;
```
Since Gordos are no longer on a timer, this can be removed

SirTanknSpank committed:
Tank forgor

Forgot to remove these lmao

SirTanknSpank committed:
Tank forgor part 2

Removing timer from const
